### DISCUSSION TITLE
How to add DoubleTapped Event to DataGridRow?

### DISCUSSION THREAD

**Discussion by iMrShadow** [AvaloniaUI/Avalonia#14769] [Q&A] [answered]
Hello! I am trying to add DoubleTapped (or something similar to MouseDoubleClick) event to a DataGridRow. 
The idea is when I am double clicking on a row, it triggers an event. It should do something similar to [this](https://stackoverflow.com/questions/22790181/wpf-datagrid-row-double-click-event-programmatically).

I already tried this method, but this also triggers when I double click the column headers, something which I want to avoid.
```c#
files_datagridview.DoubleTapped += (sender, args) =>
 {
     //do something
 };
```

This is the DataGrid section.
```xaml
<DataGrid Name="files_datagridview"
	 Margin="10,80,10,38"
	 SelectionChanged="datagridview_SelectionChanged"
	 IsReadOnly="true">

	<DataGrid.Columns>
		<DataGridTextColumn Header="File Name" Binding="{Binding FileName}" />
		<DataGridTextColumn Header="File Type" Binding="{Binding FileType}" />
	</DataGrid.Columns>
</DataGrid>
```

**Comment by timunie** [comment, marked as answer]
you may filter args.Sender for correct control type.

**Reply by iMrShadow** [reply]
Thanks! Here is the updated code segment from above
```csharp
files_datagridview.DoubleTapped += (sender, args) =>
{
    var source = args.Source;
    if (source is Border)
    {
        //do something
    }
};
```

**Reply by keynes1981** [reply]
var source = args.Source as Border;
var datagrid=sender as Datagrid;
if(source is null || datagrid is null) return;
if(source.Name=="CellBorder" )/*Tapped on content cell/*
{
     //do something
}
if(source.Name=="HeaderBackground") /*Tapped on header cell*/
{
    // do something
}
